### PR TITLE
Use createAnonymousCredentials() when AnonymousCredentialsProvider is passed in

### PR DESCRIPTION
**Pull request by davidh44**
## Motivation and Context

Adding usage of createAnonymousCredentials() to fix issue where NPE is thrown when AnonymousCredentialsProvider is passed in

Fixes https://github.com/aws/aws-sdk-java-v2/issues/3174

## Modifications

This commit adds a check of the AwsCredentialsProvider that is passed in to the constructor of CrtCredentialsProviderAdapter. If it is an instance of AnonymousCredentialsProvider, it will return createAnonymousCredentials()

## Testing
 Added tests

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [ ] Local run of `mvn install` succeeds
- [ ] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -113,7 +113,7 @@
         <rxjava.version>2.2.21</rxjava.version>
         <commons-codec.verion>1.10</commons-codec.verion>
         <jmh.version>1.29</jmh.version>
-        <awscrt.version>0.19.0</awscrt.version>
```
Comment (dagnir) [inline pom.xml:116]: Sorry just noticed this. We should add a changelog entry mentioning the version bump.

**Comment by dagnir** [comment]
`AWSCLIV2.pkg` somehow got committed. Can you remove?

File: pom.xml (hunk 1 above)
Comment (davidh44) [reply pom.xml:116]: Added entry for crt bump

**Review by dagnir — APPROVED** [review]
🎉 Nice work!

**Pull request merged**